src/main/java/uk/gov/hmcts/reform/civil/service/search/CasesStuckCheckSearchService.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.hmcts.reform.civil.service.search;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SearchResult;
+import uk.gov.hmcts.reform.civil.model.search.Query;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.index.query.QueryBuilders.rangeQuery;
+
+@Service
+@Slf4j
+public class CasesStuckCheckSearchService extends ElasticSearchService {
+
+    public CasesStuckCheckSearchService(CoreCaseDataService coreCaseDataService) {
+        super(coreCaseDataService);
+    }
+
+    @Override
+    public Set<CaseDetails> getCases() {
+        String timeNow = ZonedDateTime.now(ZoneOffset.UTC).toString();
+        SearchResult searchResult = coreCaseDataService.searchCases(query(START_INDEX, timeNow));
+
+        Set<CaseDetails> caseDetailsSet = new HashSet<>(searchResult.getCases());
+
+        int pages = calculatePages(searchResult);
+        for (int i = 1; i < pages; i++) {
+            SearchResult result = coreCaseDataService.searchCases(query(i * ES_DEFAULT_SEARCH_LIMIT, timeNow));
+            caseDetailsSet.addAll(result.getCases());
+        }
+
+        List<Long> ids = caseDetailsSet.stream().map(CaseDetails::getId).sorted().toList();
+        log.info("CasesStuckCheckSearchService: Found {} stuck case(s) in the last 7 days with ids {} at time {}", ids.size(), ids, timeNow);
+
+        return caseDetailsSet;
+    }
+
+    public Query query(int startIndex, String timeNow) {
+        log.info("Call to CasesStuckCheckSearchService query with index {} and timeNow {}", startIndex, timeNow);
+        return new Query(
+            boolQuery()
+                .should(rangeQuery("data.last_modified").lt(timeNow).gt(
+                    "now-7d"))
```

Comment:
Can we make this & 7 day limit configurable via an env var?

Reply:
sure that sounds good